core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -645,13 +645,13 @@ oppia.controller('StateResponses', [
                   $scope.trainingDataOutcomeDest = dest;
 
                   var answerGroupIndex = classificationResult.answerGroupIndex;
-                  var ruleSpecIndex = classificationResult.ruleSpecIndex;
+                  var ruleIndex = classificationResult.ruleIndex;
                   if (answerGroupIndex !==
                         _state.interaction.answerGroups.length &&
                       _state.interaction.answerGroups[
-                        answerGroupIndex].ruleSpecs[
-                          ruleSpecIndex].rule_type !==
-                            CLASSIFIER_RULESPEC_STR) {
+                        answerGroupIndex].rules[
+                          ruleIndex].ruleType !==
```

Comment:
Wow, the narrowness of the column width here is terrible. Any chance of reworking this a bit to put a bit more stuff on each line? (No worries if not -- this is totally optional!)